include/boost/astronomy/coordinate/cartesian_representation.hpp

```diff
@@ -3,132 +3,330 @@
+    template <typename Unit2, typename CoordinateType_> typename YQuantity,
+    template <typename Unit3, typename CoordinateType_> typename ZQuantity,
+    typename Unit1,
+    typename Unit2,
+    typename Unit3
+>
+cartesian_representation
+<
+    CoordinateType,
+    XQuantity<Unit1, CoordinateType>,
+    YQuantity<Unit2, CoordinateType>,
+    ZQuantity<Unit3, CoordinateType>
+> make_cartesian_representation
+(
+    XQuantity<Unit1, CoordinateType> const& x,
+    YQuantity<Unit2, CoordinateType> const& y,
+    ZQuantity<Unit3, CoordinateType> const& z
+)
+{
+    return cartesian_representation
+        <
+            CoordinateType,
+            XQuantity<Unit1, CoordinateType>,
+            YQuantity<Unit2, CoordinateType>,
+            ZQuantity<Unit3, CoordinateType>
+        >(x, y, z);
+}
+
+template
+<
+    typename ReturnCoordinateType,
+    typename ReturnXQuantity,
+    typename ReturnYQuantity,
+    typename ReturnZQuantity,
+    typename CoordinateType,
+    typename XQuantity,
+    typename YQuantity,
+    typename ZQuantity
+>
+cartesian_representation
+<
+    ReturnCoordinateType,
+    ReturnXQuantity,
+    ReturnYQuantity,
+    ReturnZQuantity
+>
+make_cartesian_representation
+(cartesian_representation<CoordinateType, XQuantity, YQuantity, ZQuantity> const& other)
+{
+    return make_cartesian_representation(
+        static_cast<ReturnXQuantity>(other.get_x()),
+        static_cast<ReturnYQuantity>(other.get_y()),
+        static_cast<ReturnZQuantity>(other.get_z())
+    );
+}
+
+template
+<
+    typename CoordinateType,
+    typename XQuantity,
+    typename YQuantity,
+    typename ZQuantity
+>
+cartesian_representation<CoordinateType, XQuantity, YQuantity, ZQuantity>
+make_cartesian_representation
+(cartesian_representation<CoordinateType, XQuantity, YQuantity, ZQuantity> const& other)
+{
+    return cartesian_representation
+        <
+            CoordinateType,
+            XQuantity,
+            YQuantity,
+            ZQuantity
+        >(other);
+}
+
+
+template
+<
+    typename CoordinateType = double,
+    typename XQuantity = bu::quantity<bu::si::dimensionless>,
+    typename YQuantity = bu::quantity<bu::si::dimensionless>,
+    typename ZQuantity = bu::quantity<bu::si::dimensionless>,
+    std::size_t OtherDimensionCount,
+    typename OtherCoordinateSystem,
+    typename OtherCoordinateType
+>
+cartesian_representation<CoordinateType, XQuantity, YQuantity, ZQuantity>
+make_cartesian_representation
+(
+    bg::model::point
+    <
+    OtherCoordinateType,
+    OtherDimensionCount,
+    OtherCoordinateSystem
+    > const& pointObject)
+{
+    return cartesian_representation
+        <
+            CoordinateType,
+            XQuantity,
+            YQuantity,
+            ZQuantity
+        >(pointObject);
+}
+
+template <typename OtherRepresentation>
+cartesian_representation
+<
+    typename OtherRepresentation::type,
+    typename OtherRepresentation::quantity3,
```

Comment:
No, this is correct because I am expecting here spherical types of coordinate. Which means first and second parameter would be in degree and only last parameter will be in length or any such quantity. As a result of conversion into cartesian, all the coordinates will be in the form of the third type of quantity from the original point/vector.